app/scripts/controllers/buildConfig.js

```diff
@@ -74,6 +109,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.buildConfig = buildConfig;
+              $scope.updatedBuildConfig = angular.copy(buildConfig);
```

Comment:
Seems like the right thing to do might be to ignore the watch while editing and let the "submit" fail because then the user would know that someone else edited/changed it already?